File: creational/builder/builder.js
class Person {
  constructor(builder) {
    this.firstName = builder.firstName;
    this.middleName = builder.middleName;
    this.lastName = builder.lastName;
    this.age = builder.age;
    this.fathersName = builder.fathersName;
    this.mothersName = builder.mothersName;
    this.height = builder.height;
    this.weight = builder.weight;
  }
}

class PersonBuilder {
  setFirstName(firstName) {
    this.firstName = firstName;
    return this;
  }

  setMiddleName(middleName) {
    this.middleName = middleName;
    return this;
  }

  setLastName(lastName) {
    this.lastName = lastName;
    return this;
  }

  setAge(age) {
    this.age = age;
    return this;
  }

  setFathersName(fathersName) {
    this.fathersName = fathersName;
    return this;
  }

  setMothersName(mothersName) {
    this.mothersName = mothersName;
    return this;
  }

  setHeight(height) {
    this.height = height;
    return this;
  }

  setWeight(weight) {
    this.weight = weight;
    return this;
  }

  build() {
    return new Person(this);
  }
}

// Creating Person with desired arguments
const Person1 = new PersonBuilder()
  .setFirstName("Nicola")
  .setLastName("Tesla")
  .build();
console.log(Person1.firstName);

File: creational/builder/builder.py
class Person:
    def __init__(self):
        pass

class PersonBuilder:

    def __init__(self):
        self.person = Person()

    def setFirstName(self, firstName):
        self.person.firstName = firstName
        return self

    def setMiddleName(self, middleName):
        self.person.middleName = middleName
        return self

    def setLastName(self, lastName):
        self.person.lastName = lastName
        return self

    def setAge(self, age):
        self.person.age = age
        return self

    def setFathersName(self, fathersName):
        self.person.mrofathersName = fathersName
        return self

    def setMothersName(self, mothersName):
        self.person.mothersName = mothersName
        return self

    def setHeight(self, height):
        self.person.height = height
        return self

    def setWeight(self, weight):
        self.person.weight = weight
        return self

    def build(self):
        return self.person


def main():
    # Creating Person with desired arguments
    person1 = PersonBuilder().setFirstName("Nicola").setLastName("Tesla").build()
    print(person1.firstName)


if __name__ == "__main__":
    main()
